### PR TITLE
Documentation/deploy-master: Explain register-node=0

### PR DESCRIPTION
**Pull request by joshix**
1. Explain why master node kubelet is configured with register-node=FALSE
2. Explain api comm fail warnings in kubelet logs on master nodes per #1, above.
3. Bump every heading up one level (naive correction of heading hierarchy)

Fixes https://github.com/coreos-inc/design/issues/165, https://github.com/coreos/coreos-kubernetes/issues/56, https://github.com/coreos/coreos-kubernetes/issues/111

### CODE REVIEW

File: Documentation/deploy-master.md [hunk 1]
```diff
@@ -74,9 +74,9 @@ After=flanneld.service
...
 
-The kubelet is the agent on each machine that starts and stops Pods and other machine-level tasks. The kubelet communicates to the API server (also running on the master machine) with the TLS certificates we placed on disk earlier.
+The [kubelet](http://kubernetes.io/v1.0/docs/admin/kubelet.html) is the agent on each machine that starts and stops Pods and other machine-level tasks. The kubelet communicates with the API server (also running on the master machine(s)) with the TLS certificates we placed on disk earlier. On the master node, the kubelet is configured to communicate with the API server, but not register for cluster work, as shown in the `--register-node=false` line in the YAML excerpt below. This prevents user pods being scheduled on the master node(s), and ensures cluster work is routed only to task-specific worker nodes. Note that the kubelet running on a master node may therefore log repeated attempts to post status to the API server. These warnings are expected behavior and can be ignored. Future Kubernetes releases plan to [handle this common deployment consideration more gracefully](https://github.com/kubernetes/kubernetes/issues/14140#issuecomment-142126864).
```
Comment (robszumski) [inline Documentation/deploy-master.md]: Might be worth it to break this into 3 paragraphs:

The kubelet...
On the master...
Note that the...

Comment (robszumski) [inline Documentation/deploy-master.md]: Should we just say "master nodes" (plural) everywhere and get rid of the `(s)`? It's going to be the main use-case and it doesn't hurt your understanding if you are only running one, IMO.

Comment (robszumski) [inline Documentation/deploy-master.md]: > Note that the kubelet running on a master node may therefore log repeated attempts to post status to the API server.

I found "therefore" to be unneeded here

"to post **its** status to the"?

**Comment by aaronlevy** [comment]
LGTM

**Pull request merged**